cities/queries/causal_insight.py

```diff
@@ -0,0 +1,413 @@
+            self.fips_id
+        ][str(year)]
+
+        self.observed_intervention_original = dg.wide[self.intervention_dataset].iloc[
+            self.fips_id
+        ][str(year)]
+
+        self.observed_outcomes = dg.std_wide[self.outcome_dataset].iloc[self.fips_id][
+            outcome_years[year_id : (year_id + 4)]
+        ]
+        self.intervention_diff = self.intervened_value - self.observed_intervention
+
+        self.intervention_impact = {}
+        self.intervention_impact_mean = []
+        self.intervention_impact_low = []
+        self.intervention_impact_high = []
+        for shift in [1, 2, 3]:
+            self.intervention_impact[shift] = (
+                self.tensed_tau_samples[shift] * self.intervention_diff
+            )
+            self.intervention_impact_mean.append(
+                np.mean(self.intervention_impact[shift])
+            )
+            self.intervention_impact_low.append(
+                np.percentile(self.intervention_impact[shift], 2.5)
+            )
+            self.intervention_impact_high.append(
+                np.percentile(self.intervention_impact[shift], 97.5)
+            )
+
+        predicted_mean = [self.observed_outcomes.iloc[0]] + (
+            self.intervention_impact_mean + self.observed_outcomes.iloc[1:]
+        ).tolist()
+        predicted_low = [self.observed_outcomes.iloc[0]] + (
+            self.intervention_impact_low + self.observed_outcomes.iloc[1:]
+        ).tolist()
+        predicted_high = [self.observed_outcomes.iloc[0]] + (
+            self.intervention_impact_high + self.observed_outcomes.iloc[1:]
+        ).tolist()
+
+        self.predictions = pd.DataFrame(
+            {
+                "year": self.prediction_years,
+                "observed": self.observed_outcomes,
+                "mean": predicted_mean,
+                "low": predicted_low,
+                "high": predicted_high,
+            }
+        )
+
+        self.predictions_original = revert_prediction_df(
+            self.predictions, self.outcome_dataset
+        )
+
+        # TODO for some reason indexing using gather doesn't pick the right indices
+        # look into this some time, do this by hand for now
+        # with self.mwc:
+        #     self.tau_samples = self.samples['weight_TY'].squeeze().detach().numpy()
+        #     self.tensed_observed_samples[shift] = self.tensed_intervened_samples[shift] = gather(
+        #     self.samples['Y'], IndexSet(**{"T": {0}}),
+        #     event_dim=0,).squeeze()
+        #     self.tensed_intervened_samples[shift] = gather(
+        #     self.samples['Y'], IndexSet(**{"T": {1}}),
+        #     event_dim=0,).squeeze()#[:,self.fips_id]
+
+        #     self.tensed_outcome_difference[shift] = (
+        #     self.tensed_intervened_samples[shift] - self.tensed_observed_samples[shift]
+        #     )
+        return
+
+    def plot_predictions(
```

Comment:
Haven't really dug into this plotting stuff (besides looking at the plots it spits out)